### DISCUSSION TITLE
Check if browser support function/method

### DISCUSSION THREAD

**Discussion by ejtipi** [fabricjs/fabric.js#7011] [Q&A]
Sometimes I get empty files from generator based on fabric.js. I figured that maybe not every browser supports all methods and function. I found something like this:

`if (!fabric.Canvas.supports('toDataURL')) alert('No');`

...but in 4.3.1 version it return "null" even if browser support this. Is there any way to test functions in script and possibly show a warning to the user?

**Comment by asturur** [comment]
which browsers that are supported by fabricJS do actually not support toDataUrl?

Can you describe how an empty file looks like?
